gateway/ln-gateway/src/lightning_manager.rs

```diff
@@ -25,6 +27,37 @@ impl std::fmt::Debug for LightningManager {
 }
 
 impl LightningManager {
+    pub fn new_initializing(lightning_builder: Arc<dyn LightningBuilder + Send + Sync>) -> Self {
+        Self {
+            lightning_builder,
+            state: Arc::new(RwLock::new(GatewayState::Initializing)),
+        }
+    }
+
+    pub async fn connect_route_htlcs(
+        &self,
+        task_group: &mut TaskGroup,
+    ) -> Result<(RouteHtlcStream, Arc<dyn ILnRpcClient>), LightningRpcError> {
+        let (stream, ln_client) = self
+            .lightning_builder
+            .build()
+            .await
+            .route_htlcs(task_group)
+            .await?;
+
+        self.set_state(GatewayState::Connected).await;
+
+        Ok((stream, ln_client))
+    }
+
+    pub async fn get_state(&self) -> GatewayState {
+        self.state.read().await.clone()
+    }
+
+    pub async fn get_state_string(&self) -> String {
```

Comment:
eh, I dont think this is necessary, its just as ergonomic to call `to_string()` after `get_state`

Reply:
Done.